File: src/osint_core/schemas/watch.py
"""Pydantic schemas for watch resources."""

import uuid
from datetime import datetime
from enum import StrEnum

from pydantic import BaseModel

from osint_core.schemas.common import PaginatedResponse


class WatchTypeEnum(StrEnum):
    persistent = "persistent"
    dynamic = "dynamic"


class WatchStatusEnum(StrEnum):
    active = "active"
    paused = "paused"
    expired = "expired"
    promoted = "promoted"


class WatchCreateRequest(BaseModel):
    name: str
    region: str | None = None
    country_codes: list[str] | None = None
    bounding_box: dict[str, float] | None = None
    keywords: list[str] | None = None
    source_filter: list[str] | None = None
    severity_threshold: str = "medium"
    plan_id: str | None = None
    ttl_hours: int | None = None


class WatchUpdateRequest(BaseModel):
    status: WatchStatusEnum | None = None
    region: str | None = None
    country_codes: list[str] | None = None
    bounding_box: dict[str, float] | None = None
    keywords: list[str] | None = None
    source_filter: list[str] | None = None
    severity_threshold: str | None = None
    ttl_hours: int | None = None


class WatchResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: uuid.UUID
    name: str
    watch_type: WatchTypeEnum
    status: WatchStatusEnum
    region: str | None = None
    country_codes: list[str] | None = None
    bounding_box: dict[str, float] | None = None
    keywords: list[str] | None = None
    source_filter: list[str] | None = None
    severity_threshold: str
    plan_id: str | None = None
    ttl_hours: int | None = None
    created_at: datetime
    expires_at: datetime | None = None
    promoted_at: datetime | None = None
    created_by: str


class WatchList(PaginatedResponse[WatchResponse]):
    pass

File: apps/web/src/types/api/watch.ts
import type { CollectionResponse, SeverityEnum, WatchStatusEnum } from './common';

export type WatchTypeEnum = 'persistent' | 'dynamic';

export interface WatchResponse {
  id: string;
  name: string;
  watch_type: WatchTypeEnum;
  status: WatchStatusEnum;
  region: string | null;
  country_codes: string[] | null;
  bounding_box: Record<string, number> | null;
  keywords: string[] | null;
  source_filter: string[] | null;
  severity_threshold: SeverityEnum;
  plan_id: string | null;
  ttl_hours: number | null;
  created_at: string;
  expires_at: string | null;
  promoted_at: string | null;
  created_by: string;
}

export type WatchList = CollectionResponse<WatchResponse>;
